framework/scripts/generate_test_list.py

```diff
@@ -74,24 +74,60 @@ def main():
 
     # generate the .cpp file
     with open(cpp_file, 'w') as f:
+        f.write('#include <set>\n')
         f.write('#include <span>\n')
-        f.write(f'#include "{h_file_basename}"\n\n')
-        f.write('using namespace std;\n\n')
+        f.write(f'#include "{h_file_basename}"\n')
         decls = [ f'extern struct test _test_{test};' for test in sorted(all_tests.keys()) ]
         f.write('\n'.join(decls))
         f.write('\n\n')
+
+        f.write('struct TestList {\n')
+        f.write('    uint64_t features;\n')
+        f.write('    const char* name;\n')
+        f.write('    const std::span<struct test* const> content;\n')
+        f.write('};\n\n')
+
         for name, tests in test_lists.items():
-            f.write(f'static constexpr struct test * const {name}_test_list[] = {{')
+            f.write(f'static constexpr struct test * const {name}_test_list_[] = {{')
             test_list = [ f'    &_test_{test}' for test in tests ]
             f.write(',\n'.join(test_list))
             f.write('\n};\n\n')
+            f.write(f'static const struct TestList {name}_test_list = {{ {"cpu_" + name if name != "default" else 0}, "{name}", {name}_test_list_ }};\n\n')
+        
+        f.write('namespace {\n')
+
+        # define sorted set of lists
+        f.write('auto isa_features_cmp = [](const TestList* lhs, const TestList* rhs) {\n')
+        f.write('    return lhs->features > rhs->features;\n')
```

Comment:
This sorting may produce false positives because the CPU features aren't added always at the end in the CPUID and our bits are therefore also all out of order.  Take for example `cpu_feature_avxvnni`, which is higher than `cpu_feature_avx512f`. That means a list for Sierra Forest would have this bit set and its numeric value would be higher than a list for Ice Lake. Then, when running on a CPU that has both features (e.g., Sapphire Rapids), we may select the SRF one instead of the ICX one.

One way around this is to sort by bit population count instead of numeric value (use [`std::popcount`](https://en.cppreference.com/w/cpp/numeric/popcount)). It may still fail, but is less likely.

Another is to not sort at all (except maybe the "default" fallback last) but instead require the maintainers to pass the lists in the proper search order.

Reply:
no sorting then